node/src/components/deploy_acceptor/tests.rs

```diff
@@ -0,0 +1,584 @@
+    fmt::{self, Debug, Display, Formatter},
+    time::Duration,
+};
+
+use derive_more::From;
+use futures::channel::oneshot;
+use prometheus::Registry;
+use reactor::ReactorEvent;
+use serde::Serialize;
+use tempfile::TempDir;
+use thiserror::Error;
+use tokio::time;
+
+use casper_execution_engine::{
+    core::engine_state::{BalanceResult, QueryResult, MAX_PAYMENT_AMOUNT},
+    storage::trie::merkle_proof::TrieMerkleProof,
+};
+use casper_types::{account::Account, CLValue, ProtocolVersion, StoredValue, URef, U512};
+
+use super::*;
+use crate::{
+    components::storage::{self, Storage},
+    effect::{
+        announcements::{ControlAnnouncement, DeployAcceptorAnnouncement},
+        requests::ContractRuntimeRequest,
+        Responder,
+    },
+    reactor::{self, EventQueueHandle, QueueKind, Runner},
+    testing::ConditionCheckReactor,
+    types::{Block, Chainspec, Deploy, NodeId},
+    utils::{Loadable, WithDir},
+    NodeRng,
+};
+
+const POLL_INTERVAL: Duration = Duration::from_millis(10);
+const TIMEOUT: Duration = Duration::from_secs(10);
+
+/// Top-level event for the reactor.
+#[derive(Debug, From, Serialize)]
+#[allow(clippy::large_enum_variant)]
+#[must_use]
+enum Event {
+    #[from]
+    Storage(#[serde(skip_serializing)] storage::Event),
+    #[from]
+    DeployAcceptor(#[serde(skip_serializing)] super::Event),
+    #[from]
+    ControlAnnouncement(ControlAnnouncement),
+    #[from]
+    DeployAcceptorAnnouncement(#[serde(skip_serializing)] DeployAcceptorAnnouncement<NodeId>),
+    #[from]
+    ContractRuntime(#[serde(skip_serializing)] ContractRuntimeRequest),
+}
+
+impl ReactorEvent for Event {
+    fn as_control(&self) -> Option<&ControlAnnouncement> {
+        if let Self::ControlAnnouncement(ref ctrl_ann) = self {
+            Some(ctrl_ann)
+        } else {
+            None
+        }
+    }
+}
+
+impl From<StorageRequest> for Event {
+    fn from(request: StorageRequest) -> Self {
+        Event::Storage(storage::Event::from(request))
+    }
+}
+
+impl Display for Event {
+    fn fmt(&self, formatter: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Event::Storage(event) => write!(formatter, "storage: {}", event),
+            Event::DeployAcceptor(event) => write!(formatter, "deploy acceptor: {}", event),
+            Event::ControlAnnouncement(ctrl_ann) => write!(formatter, "control: {}", ctrl_ann),
+            Event::DeployAcceptorAnnouncement(ann) => {
+                write!(formatter, "deploy-acceptor announcement: {}", ann)
+            }
+
+            Event::ContractRuntime(event) => {
+                write!(formatter, "contract-runtime event: {:?}", event)
+            }
+        }
+    }
+}
+
+/// Error type returned by the test reactor.
+#[derive(Debug, Error)]
+enum Error {
+    #[error("prometheus (metrics) error: {0}")]
+    Metrics(#[from] prometheus::Error),
+}
+
+#[allow(clippy::enum_variant_names)]
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+enum TestScenario {
```

Comment:
Can you consider extracting this enum to a separate file? The `tests.rs` is already quite long and the `TestScenario` comes with an moderately long `impl` block.

Reply:
I am currently fine with having it being present in the same file, however, I feel with follow up work on the acceptor, the test suite may grow and then it may make sense to revisit this concern